libndt7/src/main/java/net/measurementlab/ndt/android/Client.java

```diff
@@ -0,0 +1,251 @@
+    }
+
+    @Override
+    public final void onMessage(WebSocket ws,
+                                String text) {
+        onLogInfo("WebSocket onMessage");
+        count += text.length();
+        periodic();
+
+        @NonNull
+        final JSONObject doc;
+
+        // Root
+        try {
+            doc = new JSONObject(text);
+        } catch (JSONException e) {
+            Log.d(TAG, "could not parse root json", e);
+            return;
+        }
+
+        final double elapsed;
+
+        // Top level
+        try {
+            elapsed = doc.getDouble("elapsed");
+        } catch (JSONException e) {
+            Log.e(TAG, "did not find valid elapsed time", e);
+            return;
+        }
+
+        @Nullable
+        Measurement.TcpInfo tcpInfo = null;
+
+        // tcp_info
+        try {
+            JSONObject tcpInfoObject = doc.getJSONObject("tcp_info");
+
+            double smoothedRtt = tcpInfoObject.getDouble("smoothed_rtt");
+            double rttVar = tcpInfoObject.getDouble("rtt_var");
+            tcpInfo = new Measurement.TcpInfo(smoothedRtt, rttVar);
+        } catch (JSONException e) {
+            Log.d(TAG, "did not find valid tcp_info", e);
+        }
+
+        @Nullable
+        Measurement.BBRInfo bbrInfo = null;
+
+        // bbr_info
+        try {
+            JSONObject bbrInfoObject = doc.getJSONObject("bbr_info");
+            long bandwidth = bbrInfoObject.getLong("max_bandwidth");
+            double minRtt = bbrInfoObject.getDouble("min_rtt");
+            bbrInfo = new Measurement.BBRInfo(bandwidth, minRtt);
+        } catch (JSONException e) { }
+
+        @Nullable
+        Measurement.AppInfo appInfo = null;
+
+        // app_info
+        try {
+            JSONObject appInfoObject = doc.getJSONObject("app_info");
+            appInfo = new Measurement.AppInfo(appInfoObject.getLong("num_bytes"));
+        } catch (JSONException e) { }
+
+        Measurement measurement = new Measurement(elapsed, tcpInfo, bbrInfo, appInfo);
+        onServerDownloadMeasurement(measurement);
+    }
+
+    @Override
+    public final void onMessage(WebSocket ws,
+                                ByteString bytes) {
+        count += bytes.size();
+        periodic();
+    }
+
+    @Override
+    public final void onClosing(WebSocket ws,
+                                int code,
+                                String reason) {
+        // TODO(bassosimone): make sure code has the correct value otherwise
+        // we must return an error to the caller.
+        ws.close(1000, null);
+    }
+
+    @Override
+    public final void onFailure(WebSocket ws,
+                                Throwable t,
+                                Response r) {
+        onError(t.getMessage());
+        rv = false;
+    }
+
+    @CheckResult
+    public boolean runDownload() {
+        URI uri;
+        try {
+            uri = new URI(
+                (settings.disableTLS) ? "ws" : "wss",
```

Comment:
As mentioned `disableTLS` should not be a setting and we should always use `wss`